ginivision/src/main/java/net/gini/android/vision/analysis/AnalysisFragmentImpl.java

```diff
@@ -287,26 +333,36 @@ public void onGlobalLayout() {
     }
 
     private void onViewLayoutFinished() {
-        rotateDocumentImageView();
+        showDocument();
         analyzeDocument();
     }
 
-    private void rotateDocumentImageView() {
+    private void rotateDocumentImageView(final int rotationForDisplay) {
         int newWidth = mLayoutRoot.getWidth();
         int newHeight = mLayoutRoot.getHeight();
-        if (mPhoto.getRotationForDisplay() == 90 || mPhoto.getRotationForDisplay() == 270) {
+        if (rotationForDisplay == 90 || rotationForDisplay == 270) {
             newWidth = mLayoutRoot.getHeight();
             newHeight = mLayoutRoot.getWidth();
         }
 
-        FrameLayout.LayoutParams layoutParams = (FrameLayout.LayoutParams) mImageDocument.getLayoutParams();
+        final FrameLayout.LayoutParams layoutParams = (FrameLayout.LayoutParams) mImageDocument.getLayoutParams();
         layoutParams.width = newWidth;
         layoutParams.height = newHeight;
         mImageDocument.setLayoutParams(layoutParams);
-        mImageDocument.setRotation(mPhoto.getRotationForDisplay());
+        mImageDocument.setRotation(rotationForDisplay);
     }
 
     private void showDocument() {
-        mImageDocument.setImageBitmap(mPhoto.getBitmapPreview());
+        final Size previewSize = new Size(mImageDocument.getWidth(), mImageDocument.getHeight());
+        mDocumentRenderer.toBitmap(previewSize, new DocumentRenderer.Callback() {
```

Comment:
Abstracted away document rendering since we either decode an image into a bitmap or render a pdf into a bitmap.